rest/models.py

```diff
@@ -12,3 +13,64 @@ def __init__(self, response: requests.Response):
         dict_rsp = json.loads(response.content)
         self.buy = dict_rsp.get('buy')
         self.sell = dict_rsp.get('sell')
+
+class Ticker:
+    high: decimal.Decimal
+    last: decimal.Decimal
+    timestamp: str
+    bid: decimal.Decimal
+    vwap: decimal.Decimal
+    volume: decimal.Decimal
+    low: decimal.Decimal
+    ask: decimal.Decimal
+    open: decimal.Decimal
+
+    def __init__(self, response: requests.Response):
+        dict_rsp = json.loads(response.content)
+        self.high = dict_rsp.get('high')
+        self.last = dict_rsp.get('last')
+        self.timestamp = dict_rsp.get('timestamp')
+        self.bid = dict_rsp.get('bid')
+        self.vwap = dict_rsp.get('vwap')
+        self.volume = dict_rsp.get('volume')
+        self.low = dict_rsp.get('low')
+        self.ask = dict_rsp.get('ask')
+        self.open = dict_rsp.get('open')
+
+class HourlyTicker:
+    high: decimal.Decimal
+    last: decimal.Decimal
+    timestamp: str
+    bid: decimal.Decimal
+    vwap: decimal.Decimal
+    volume: decimal.Decimal
+    low: decimal.Decimal
+    ask: decimal.Decimal
+    open: decimal.Decimal
+
+    def __init__(self, response: requests.Response):
+        dict_rsp = json.loads(response.content)
+        self.high = dict_rsp.get('high')
+        self.last = dict_rsp.get('last')
+        self.timestamp = dict_rsp.get('timestamp')
+        self.bid = dict_rsp.get('bid')
+        self.vwap = dict_rsp.get('vwap')
+        self.volume = dict_rsp.get('volume')
+        self.low = dict_rsp.get('low')
+        self.ask = dict_rsp.get('ask')
+        self.open = dict_rsp.get('open')
+
+
+class OrderBook:
+    timestamp: str
+    microtimestamp: str
+    bids: typing.List
```

Comment:
This is incorrect type for bids and asks. Should be at least `typing.List[typing.List[[str]]`. However, each list includes 2 things.
Price and Amount. hence, I suggest better modeling.

You see, now we have issues, as someone using order book entry doesn't have much use for it. Has to loop over 2 grades of lists and .... uugh, ugly as hell.

Lets use some named tuples!
We add one import and define our named tuple at the top of this file. Something like:
```
from collections import namedtuple

OrderBookEntry = namedtuple('OrderBookEntry', 'price amount')
```

OrderBook class should then have private instance method.

```
def _serialize_order_book_entries(entries: typing.List[typing.List[str]]) -> typing.List[OrderBookEntry]:
         serialized = []
         for entry in entries:
                serialized.append(OrderBookEntry(entry[0], entry[1])

        return serialized
```
We now have modeling that we want. in constructor you now call something like:

```
self.bids = self._serialize_order_book_entries(dict_rsp.get('bids'))
self.asks = self._serialize_order_book_entries(dict_rsp.get('asks'))
```
Then someone can use this order book much easier.
Like: `order_book.bids[0].price`

Reply:
Ough, I wrote this code without trying it out, maybe some typos or so. check it out.

Reply:
This has been placed in case of order book and evolved.
Resolving as I see no futher conflict.